### PR TITLE
Batches implementation

### PR DESCRIPTION
**Pull request by Mingela**
Added iroha batches interaction interface

Added reduced payload hash computing in Utils

Added a test in the main integration workflow

https://github.com/Warchant/iroha-pure-java/issues/39

### CODE REVIEW

**Comment by Mingela** [comment]
Would appreciate any ideas about FieldValidator updates if needed

File: src/test/groovy/jp/co/soramitsu/iroha/java/IntegrationTest.groovy [hunk 1]
```diff
@@ -128,5 +129,59 @@ class IntegrationTest extends Specification {
...
+        ]
+        def batch_reduced_hashes = [
+                Utils.toHex(Utils.reducedHash(batchPrototype.get(0))),
+                Utils.toHex(Utils.reducedHash(batchPrototype.get(1))),
+                Utils.toHex(Utils.reducedHash(batchPrototype.get(2)))
+        ]
+        def batch = [
+                Transaction.builder(defaultAccountId, now)
+                        .createAccount("${anotherAccount}", defaultDomain, defaultKeypair.getPublic())
+                        .setBatchMeta(TransactionOuterClass.Transaction.Payload.BatchMeta.BatchType.ORDERED, batch_reduced_hashes)
```
Comment (Warchant) [inline src/test/groovy/jp/co/soramitsu/iroha/java/IntegrationTest.groovy]: use static import

File: src/test/groovy/jp/co/soramitsu/iroha/java/IntegrationTest.groovy [hunk 2]
```diff
@@ -128,5 +129,59 @@ class IntegrationTest extends Specification {
...
+                        .build(),
+                Transaction.builder(defaultAccountId, now)
+                        .setAccountDetail(anotherAccountId, "key", "value")
+                        .sign(defaultKeypair)
+                        .build()
+        ]
+        def batch_reduced_hashes = [
+                Utils.toHex(Utils.reducedHash(batchPrototype.get(0))),
+                Utils.toHex(Utils.reducedHash(batchPrototype.get(1))),
+                Utils.toHex(Utils.reducedHash(batchPrototype.get(2)))
```
Comment (Warchant) [inline src/test/groovy/jp/co/soramitsu/iroha/java/IntegrationTest.groovy]: add `String Transaction.reducedHashHex()` and then replace with 

```
def batch_reduced_hashes = batchPrototype.stream().map(Transaction.&reducedHashHex)
.collect(Collectors.toList());
```

File: src/main/java/jp/co/soramitsu/iroha/java/Transaction.java [hunk 3]
```diff
@@ -51,6 +66,24 @@
...
+  @Override
+  public String getReducedHashHex() {
+    return Utils.toHex(getReducedHash());
+  }
+
+  @Override
+  public TransactionBuilder makeMutable() {
+    while (tx.getSignaturesCount() > 0) {
+      tx.removeSignatures(0);
+    }
```
Comment (Warchant) [inline src/main/java/jp/co/soramitsu/iroha/java/Transaction.java]: use clearSignatures()

File: src/main/java/jp/co/soramitsu/iroha/java/Utils.java [hunk 4]
```diff
@@ -105,6 +119,34 @@ public static TxList createTxList(Iterable<TransactionOuterClass.Transaction> li
...
+    return createBatch(list, BatchType.ORDERED, keyPair);
+  }
+
+  public static Iterable<TransactionOuterClass.Transaction> createTxAtomicBatch(
+      Iterable<TransactionOuterClass.Transaction> list, KeyPair keyPair) {
+    return createBatch(list, BatchType.ATOMIC, keyPair);
+
+  }
+
+  private static Iterable<String> getBatchHashes(Iterable<TransactionOuterClass.Transaction> list) {
```
Comment (Warchant) [inline src/main/java/jp/co/soramitsu/iroha/java/Utils.java]: rename to getBatchHashesHex

**Review by Warchant — APPROVED** [review]
lets wait for CI and checks, then merge

**Pull request merged**